sdk/python/feast/base_feature_view.py

```diff
@@ -73,12 +73,8 @@ def __init__(
         Raises:
             ValueError: A field mapping conflicts with an Entity or a Feature.
         """
-        if not name:
-            raise ValueError("Name needs to be provided")
```

Comment:
I think it's a bit unnecessary since this initialization happens only internally; the check should still happen so I'm going to switch to an `assert` (since we expect it to always pass)